internal/authn/keychain.go

```diff
@@ -28,17 +30,21 @@ func WithDockerConfig(dockercfg string) PreflightKeychainOption {
 	}
 }
 
-var keychain = preflightKeychain{}
+var keychain = preflightKeychain{
+	ctx: context.Background(), // Initialize here, but can be overridden with PreflightKeychain func
+}
 
 // PreflightKeychain will return the preflight keychain as a craneauthn.Keychain.
 // This operates as a singleton. If provided an option, that option overwrites
 // the single instance of PreflightKeychain. If provided no option, the keychain
 // is returned as already configured.
-func PreflightKeychain(opts ...PreflightKeychainOption) craneauthn.Keychain {
+func PreflightKeychain(ctx context.Context, opts ...PreflightKeychainOption) craneauthn.Keychain {
 	for _, opt := range opts {
 		opt(&keychain)
 	}
 
+	keychain.ctx = ctx
```

Comment:
Should we check `ctx` for nil here? Otherwise, there's no real point in initializing a few lines up because it will _always_ get replaced here.

Reply:
I initialized it to support the tests. There's really no harm in initializing the default then always replacing with the supplied one here.